pymatgen/io/vasp/sets.py

```diff
@@ -3179,3 +3190,57 @@ def batch_write_input(
     [[0, 0, 0]],
     site_properties={"magmom": [[0, 0, 1]]},
 )
+
+
+def get_valid_magmom_struct(structure, inplace=True, spin_mode="auto"):
```

Comment:
Function is here

Reply:
Can you clarify when this happens? Is this for when some site properties are missing or `None`? Not sure if an easier solution might be to cast to the `Magmom` class.

Reply:
Both cases can happen but the missing properties case is probably more common I flagged a case in the tests that occur pretty often.  It can be avoided but it just bothers me that no error is raised until the job is submitted and running.

Also, my understanding is that Magmom class is focused on SOC stuff.

Reply:
BTW the pre-commit stuff is making these reviews a bit nasty... do you wanna run the pre-commit on the entire codebase at some point or just do it piece by piece like this?

Or should we have some kind of simplified workflow like turn off pre-commit until the end right before merge??

Again sorry for the mess.

Reply:
Yup, I was actually intending to do that soon anyway, stay tuned ...

> Or should we have some kind of simplified workflow like turn off pre-commit until the end right before merge??

Is this possible?

Reply:
> Also, my understanding is that Magmom class is focused on SOC stuff.

Magmom class is an attempt to reconcile vector vs scalar magmoms, e.g. if you run without SOC it will cast to a float, via float(Magmom) -> scalar.

I'm curious how missing properties actually happen in practice however. Guessing it's a case where you insert an additional site?

Reply:
Yeah that is the most common case for me, but there are other times when sites get copied and mutate where you can loose data.
Since I don't think we can add validation in all those places this seems like a good place for a sanity check

Reply:
Ok, have run linters module-wide, it has introduced a few merge conflicts though mostly in module sort order.

Reply:
Awesome. I saw that.
Does this mean I have to redo this PR?

Reply:
NVM, I just merge in the new master.

Reply:
Yeah, or you can fix merge conflicts on GitHub.com too, usually only takes a minute or two